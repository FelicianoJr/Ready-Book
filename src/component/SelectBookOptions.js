import React from 'react'

const SelectBookOptions = ({ book, update }) => {
    
    return (
        <select value={book.shelf}
            onChange={(event) => update(book, event.target.value)}>
            <option value="" disabled>Move to...</option>
            <option value="currentlyReading">Currently Reading</option>
            <option value="wantToRead">Want to Read</option>
            <option value="read">Read</option>
            <option value="">None</option>
        </select>
    )
};

export default SelectBookOptions;