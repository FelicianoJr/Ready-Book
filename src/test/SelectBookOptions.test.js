import SelectBookOptions from '../component/SelectBookOptions'
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("<SelectBookOptions/>", () => {

    let update;
    let props;
    let wrapper;

    beforeEach(() => {
        update = jest.fn();
        props = {
            book: {
                id: 0, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React', shelf: 'read'
            },
            update: update,
        };
        wrapper = shallow(<SelectBookOptions {...props} />)
    })

    it('should render one <select>', () => {
        expect(wrapper.find('select').length).toEqual(1);
    });

    it("when simulating a change, select should update its value", () => {
        wrapper.find('select').simulate('change', { target: { value: '100' } });
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(update).toHaveBeenCalledTimes(1)
    });

    it('should render five <option>', () => {
        expect(wrapper.find('option').length).toEqual(5);
    });

    it('should render text node <option>', () => {
        const text = wrapper.find('option').map(node => node.text());
        expect(text).toEqual(['Move to...', 'Currently Reading', 'Want to Read', 'Read', 'None'])
    })

})