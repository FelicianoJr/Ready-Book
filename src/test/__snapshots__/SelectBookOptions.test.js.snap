exports[`<SelectBookOptions/> when simulating a change, select should update its value 1`] = `
<select
  onChange={[Function]}
  value="read">
  <option
    disabled={true}
    value="">
    Move to...
  </option>
  <option
    value="currentlyReading">
    Currently Reading
  </option>
  <option
    value="wantToRead">
    Want to Read
  </option>
  <option
    value="read">
    Read
  </option>
  <option
    value="">
    None
  </option>
</select>
`;
